examples/mandelbrot/main.zig

```diff
@@ -0,0 +1,252 @@
+        .usage = .{ .vertex = true },
+        .size = @sizeOf(Vertex) * vertices.len,
+        .mapped_at_creation = true,
+    });
+    var vertex_mapped = vertex_buffer.getMappedRange(Vertex, 0, vertices.len);
+    std.mem.copy(Vertex, vertex_mapped, vertices[0..]);
+    vertex_buffer.unmap();
+    defer vertex_buffer.release();
+
+    const index_buffer = app.device.createBuffer(&.{
+        .usage = .{ .index = true },
+        .size = @sizeOf(@TypeOf(indices[0])) * indices.len,
+        .mapped_at_creation = true,
+    });
+    var index_mapped = index_buffer.getMappedRange(@TypeOf(indices[0]), 0, indices.len);
+    std.mem.copy(@TypeOf(indices[0]), index_mapped, indices[0..]);
+    index_buffer.unmap();
+    defer index_buffer.release();
+
+    const uniform_buffer = app.device.createBuffer(&.{
+        .usage = .{ .copy_dst = true, .uniform = true },
+        .size = @sizeOf(UniformBufferObject),
+        .mapped_at_creation = false,
+    });
+    defer uniform_buffer.release();
+    const bind_group = app.device.createBindGroup(
+        &gpu.BindGroup.Descriptor{
+            .layout = bgl,
+            .entries = &.{
+                gpu.BindGroup.Entry.buffer(0, uniform_buffer, 0, @sizeOf(UniformBufferObject)),
+            },
+        },
+    );
+    defer bind_group.release();
+
+    ctx.* = FrameParams{
+        .pipeline = app.device.createRenderPipeline(&pipeline_descriptor),
+        .queue = queue,
+        .vertex_buffer = vertex_buffer,
+        .index_buffer = index_buffer,
+        .uniform_buffer = uniform_buffer,
+        .bind_group = bind_group,
+    };
+
+    vs_module.release();
+    fs_module.release();
+    pipeline_layout.release();
+    bgl.release();
+
+    try app.run(.{ .frame = frame });
+}
+
+const FrameParams = struct {
+    pipeline: gpu.RenderPipeline,
+    queue: gpu.Queue,
+    vertex_buffer: gpu.Buffer,
+    index_buffer: gpu.Buffer,
+    uniform_buffer: gpu.Buffer,
+    bind_group: gpu.BindGroup,
+};
+
+fn frame(app: *App, params: *FrameParams) !void {
+    const back_buffer_view = app.swap_chain.?.getCurrentTextureView();
+    const color_attachment = gpu.RenderPassColorAttachment{
+        .view = back_buffer_view,
+        .resolve_target = null,
+        .clear_value = std.mem.zeroes(gpu.Color),
+        .load_op = .clear,
+        .store_op = .store,
+    };
+
+    const encoder = app.device.createCommandEncoder(null);
+    const render_pass_info = gpu.RenderPassEncoder.Descriptor{
+        .color_attachments = &.{color_attachment},
+        .depth_stencil_attachment = null,
+    };
+
+    {
+        const time = @intToFloat(f32, timer.read()) / @as(f32, std.time.ns_per_s);
+        const model = zm.rotationZ(0);
+        const view = zm.lookAtRh(
+            zm.f32x4(0, 2, 0, 1),
+            zm.f32x4(0, 0, 0, 1),
+            zm.f32x4(0, 0, 1, 0),
+        );
+        const proj = zm.perspectiveFovRh(
+            (std.math.pi / 7.0), //(std.math.sin(time) * 7 + 9)),
```

Comment:
Stray commented out code?

Reply:
I felt like leaving It so that when viewing the example you had an Easy way to zoom in, but i think I should probably add later the zoom from mouse input instead of time

Reply:
I like it 👍 It'd be great to add zoom/panning via mouse input in the future (not as part of this PR.)